pkg/v3/resource/namespace/current.go

```diff
@@ -0,0 +1,61 @@
+package namespace
+
+import (
+	"context"
+
+	"github.com/giantswarm/cluster-operator/pkg/v3/guestcluster"
+	"github.com/giantswarm/cluster-operator/pkg/v3/key"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	apiv1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	objectMeta, err := r.toClusterObjectMetaFunc(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Guest cluster namespace is not deleted so cancel the reconcilation. The
+	// namespace will be deleted when the guest cluster resources are deleted.
+	if key.IsDeleted(objectMeta) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling namespace deletion: deleted with the guest cluster")
+		resourcecanceledcontext.SetCanceled(ctx)
+		return nil, nil
+	}
+
+	guestK8sClient, err := r.getGuestK8sClient(ctx, obj)
+	if guestcluster.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "did not get a K8s client for the guest cluster")
+
+		// We can't continue without a K8s client. We will retry during the
+		// next execution.
+		resourcecanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource reconciliation for custom object")
+
+		return nil, nil
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
```

Comment:
We also need to cancel the reconciliation if we can't get a k8s client for the guest cluster. This can happen during the delay while the guest cluster is provisioned.